### ISSUE TITLE
Migrate S1944: Remove the SE part

### ISSUE THREAD

**Issue by pavel-mikula-sonarsource** [SonarSource/sonar-dotnet#7301]
Migrate S1944 `InvalidCastToInterfaceSymbolicExecution` to the new engine by removing the SE part from it.

The relevant case is reported by S3655 which is a more suitable rule for this detection.
```
int? value = null;
_ = (int)value;
```
